### DISCUSSION TITLE
MUI5 Data Grid Table - Displaying Multiple Columns in a Single Table without Horizontal Scroll

### DISCUSSION THREAD

**Discussion by Kavin-Brat** [community/community#102959] [Programming Help]
### Body

I'm currently working with the MUI5 Data Grid Table to present multiple columns of data. The table includes a horizontal scroll for navigating through the columns, but the customer is requesting a layout where all columns are visible without the need for horizontal scrolling.

I've attempted adjusting column widths, reducing column headers, and decreasing font sizes, but these adjustments only go so far. I'm looking for suggestions on how to handle this situation. One idea I'm considering is implementing a full-screen view.

Any insights or alternative ideas on how to achieve a clean presentation of multiple columns in a single view without relying on horizontal scrolling?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by amankr1469** [comment]
The data grid table is contained within a div element with the class data-grid, which has overflow-x: auto;,CSS property to allow horizontal scrolling when needed.
The table is designed to be responsive, meaning it adjusts its layout based on the screen size.
Using media queries, columns that are less important or may not fit well on smaller screens are hidden.

**Reply by Kavin-Brat** [reply]
@amankr1469 Even though the datagrid is responsive, each column retains some space based on the data, imposing a limit on the screen size it can accommodate. However, to eliminate the need for a scrollbar, the inquiry is about determining the optimal number of columns. This is why I opted for a full-screen view.
